### DISCUSSION TITLE
Using Numpy Vectors as inputs into PyLTSpice

### DISCUSSION THREAD

**Discussion by chekfung** [nunobrum/PyLTSpice#81] [General]
Since the LTSpice inputs are kind of wonky, I was wondering whether it would be possible to generate numpy vectors (time and signal) containing the input signals into a circuit. 

I am trying to simulate many different inputs through the same circuit using this method, but want to have python generate the signals because the input signals are fairly complex and governed by many math equations. Any suggestion on how I might get this functionality, or is this not currently supported?

Thanks!

**Comment by nunobrum** [comment]
Hello Jason

There are two ways I can think of. Either using a PWL file, or using a WAV file.
Not knowing the size of your vector, I don't know which one is best. 
Here are some useful links for your reference.

https://www.analog.com/en/analog-dialogue/raqs/raq-issue-175.html
https://www.analog.com/en/technical-articles/ltspice-importing-exporting-pwl-data.html

I haven't coded any support for this in LTSpice, although, I believe you can quickly code something that does it.
If you do, please feel welcome to push it back into this repository, or just post it here for other users to use it.

Best Regards,
Nuno
